cooltools/insulation.py

```diff
@@ -399,12 +471,16 @@ def _find_insulating_boundaries_dense(
             )
         )
 
-    ins_chrom_tables = []
-    for chrom in chromosomes:
-        ins_chrom = clr.bins().fetch(chrom)[["chrom", "start", "end"]]
-        is_bad_bin = np.isnan(clr.bins().fetch(chrom)["weight"].values)
+    ins_region_tables = []
+    for chrom, start, end, name in view_df[['chrom', 'start', 'end', 'name']].values:
+        region = [chrom, start, end]
+        ins_region = clr.bins().fetch(region)[["chrom", "start", "end"]]
+        is_bad_bin = np.isnan(clr.bins().fetch(region)[clr_weight_name].values)
 
-        m = clr.matrix(balance=balance).fetch(chrom)
+        if balance==True:
```

Comment:
I've removed the balance option - to make API aligned with other tools that we have